### PR TITLE
PSP-7271 : Create/Edit Offer (Disposition)

### PR DESCRIPTION
**Pull request by eddherrera**
image

### CODE REVIEW

**Comment by devinleighsmith** [comment]
(Grammar)
An Offer with 'Accepted' status already exist.
->
An Offer with 'Accepted' status already exists.

File: source/backend/api/Services/DispositionFileService.cs [hunk 1]
```diff
@@ -122,6 +124,59 @@ public IEnumerable<PimsDispositionOffer> GetOffers(long dispositionFileId)
...
+            {
+                throw new BadRequestException("Invalid dispositionFileId.");
+            }
+
+            return _dispositionFileRepository.GetDispositionOfferById(dispositionFileId, dispositionOfferId);
+        }
+
+        public PimsDispositionOffer AddDispositionFileOffer(long dispositionFileId, PimsDispositionOffer dispositionOffer)
+        {
+            _logger.LogInformation("Getting disposition file offers with DispositionFileId: {id}", dispositionFileId);
```
Comment (devinleighsmith) [inline source/backend/api/Services/DispositionFileService.cs:144]: nit: log message is not correct.

File: source/backend/apimodels/Models/Concepts/DispositionFile/DispositionFileOfferModel.cs [hunk 2]
```diff
@@ -8,7 +8,7 @@ public class DispositionFileOfferModel : BaseConcurrentModel
         /// <summary>
         /// get/set - The relationship id.
         /// </summary>
-        public long Id { get; set; }
```
Comment (devinleighsmith) [inline source/backend/apimodels/Models/Concepts/DispositionFile/DispositionFileOfferModel.cs:11]: is this actually nullable?

File: source/frontend/src/features/mapSideBar/disposition/router/DispositionRouter.tsx [hunk 3]
```diff
@@ -53,16 +55,31 @@ export const DispositionRouter: React.FC<IDispositionRouterProps> = props => {
         </Route>
         <AppRoute
           exact
-          path={`${stripTrailingSlash(path)}/${FileTabType.OFFERS_AND_SALE}/addOffer`}
+          path={`${stripTrailingSlash(path)}/${FileTabType.OFFERS_AND_SALE}/offers/new`}
```
Comment (devinleighsmith) [inline source/frontend/src/features/mapSideBar/disposition/router/DispositionRouter.tsx:58]: yes, this is better.

File: source/frontend/src/features/mapSideBar/disposition/tabs/offersAndSale/dispositionOffer/add/AddDispositionOfferContainer.tsx [hunk 4]
```diff
@@ -1,11 +1,76 @@
...
-> = ({ dispositionFileId }) => {
-  return <p>Add Disposition Offer Works!</p>;
+> = ({ dispositionFileId, View }) => {
+  const history = useHistory();
+  const location = useLocation();
+
+  const { setModalContent, setDisplayModal } = useModalContext();
+  const [offerStatusError, setOfferStatusError] = useState(false);
+
+  const backUrl = location.pathname.split('/offers/new')[0];
```
Comment (devinleighsmith) [inline source/frontend/src/features/mapSideBar/disposition/tabs/offersAndSale/dispositionOffer/add/AddDispositionOfferContainer.tsx:28]: I'm not sure I love this pattern. Is this what we are doing in acquisition?

File: source/frontend/src/features/mapSideBar/disposition/tabs/offersAndSale/dispositionOffer/add/AddDispositionOfferContainer.tsx [hunk 5]
```diff
@@ -1,11 +1,76 @@
...
+    postDispositionFileOffer: { execute: postDispositionOffer, loading },
+  } = useDispositionProvider();
+  const initialValues = new DispositionOfferFormModel(null, dispositionFileId);
+
+  const handleSave = async (newOffer: Api_DispositionFileOffer) => {
+    setOfferStatusError(false);
+    return postDispositionOffer(dispositionFileId, newOffer);
+  };
+
+  const handleSucces = async () => {
```
Comment (devinleighsmith) [inline source/frontend/src/features/mapSideBar/disposition/tabs/offersAndSale/dispositionOffer/add/AddDispositionOfferContainer.tsx:39]: nit: success

**Review by devinleighsmith — APPROVED** [review]

File: source/backend/apimodels/Models/Concepts/DispositionFile/DispositionFileOfferModel.cs (hunk 2 above)
Comment (eddherrera) [reply source/backend/apimodels/Models/Concepts/DispositionFile/DispositionFileOfferModel.cs:11]: For new offers the value comes as null.

File: source/frontend/src/features/mapSideBar/disposition/tabs/offersAndSale/dispositionOffer/add/AddDispositionOfferContainer.tsx (hunk 4 above)
Comment (eddherrera) [reply source/frontend/src/features/mapSideBar/disposition/tabs/offersAndSale/dispositionOffer/add/AddDispositionOfferContainer.tsx:28]: It was used for Form8

File: source/frontend/src/features/mapSideBar/disposition/tabs/offersAndSale/dispositionOffer/add/AddDispositionOfferContainer.tsx (hunk 5 above)
Comment (eddherrera) [reply source/frontend/src/features/mapSideBar/disposition/tabs/offersAndSale/dispositionOffer/add/AddDispositionOfferContainer.tsx:39]: Updated

File: source/backend/api/Services/DispositionFileService.cs (hunk 1 above)
Comment (eddherrera) [reply source/backend/api/Services/DispositionFileService.cs:144]: Updated

**Pull request merged**